### PR TITLE
Update `selectable_school` on provider model from TTAPI

### PR DESCRIPTION
**Pull request by inulty-dfe**
## Context

A new property `selectable_school` on the Provider model needs to be populated / updated from the TTAPI.

## Changes proposed in this pull request

Update the `Provider#selectable_school` from the TTAPI

## Things to check

- [ ] If the code removes any existing feature flags, a data migration has also been added to delete the entry from the database
- [ ] This code does not rely on migrations in the same Pull Request
- [ ] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [ ] If this code adds a column to the DB, decide whether it needs to be in analytics yml file or analytics blocklist
- [x] API release notes have been updated if necessary
- [ ] If it adds a significant user-facing change, is it documented in the [CHANGELOG](CHANGELOG.md)?
- [ ] Required environment variables have been updated [added to the Azure KeyVault](/docs/environment-variables.md#deploy-pipeline)
- [ ] Inform data insights team due to database changes
- [x] Make sure all information from the Trello card is in here
- [x] Rebased main
- [x] Cleaned commit history
- [x] Tested by running locally
- [ ] Add PR link to Trello card

### CODE REVIEW

**Comment by inulty-dfe** [comment]
[Update `selectable school` on provider model from API](https://trello.com/c/g2mKJxGs/2216-update-selectable-school-on-provider-model-from-api)

File: spec/examples/teacher_training_api/provider_list_response.json [hunk 1]
```diff
@@ -1,9 +1,11 @@
 {
   "data": [
     {
-      "id": 1,
```
[the remark this answers is not in the document]
Comment (inulty-dfe) [reply spec/examples/teacher_training_api/provider_list_response.json:4]: Yeah, the server responds with this as a string

[the remark this answers is not in the document]
Comment (inulty-dfe) [reply spec/examples/teacher_training_api/provider_list_response.json:4]: We should have a system that updates the examples with new properties that we add to the API. I'll create a dev ticket in the backlog

[the remark this answers is not in the document]
Comment (dcyoung-dev) [reply spec/examples/teacher_training_api/provider_list_response.json:4]: The [VCR gem](https://github.com/vcr/vcr) is pretty good at this

**Review by dcyoung-dev — COMMENTED** [review]
Do we have any specs covering the SyncProvider?

**Comment by inulty-dfe** [comment]
> Do we have any specs covering the SyncProvider?

We've got these tests
https://github.com/DFE-Digital/apply-for-teacher-training/blob/main/spec/services/teacher_training_public_api/sync_provider_spec.rb

**Pull request merged**